exp/api/v1alpha4/azuremanagedmachinepool_webhook.go

```diff
@@ -44,6 +44,10 @@ func (r *AzureManagedMachinePool) Default(client client.Client) {
 		r.Labels = make(map[string]string)
 	}
 	r.Labels[LabelAgentPoolMode] = r.Spec.Mode
+
+	if r.Spec.Name == nil {
```

Comment:
just to be safe
```suggestion
	if r.Spec.Name == nil  || r.Spec.Name == ""{
```
Not sure

Reply:
Can we also add/update tests for this?

Reply:
Sure. Added test for this.

I left the defaulting logic as is, but added validation logic to check if `Spec.Name == ""`

Reply:
I see what you're doing in `validate` but AFAIK the general way of handling empty strings in defaults has been to just assign a default like [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7f78c7bc945b8cabc35c476c09013fb1a9dab320/api/v1alpha4/azurecluster_default.go#L77) for example. I'll leave to other reviewers on this.

Reply:
oh i see, makes sense. Update the code according to your suggestion. Squashed the commits as well. Thanks for reviewing the PR @shysank !